solr/core/src/java/org/apache/solr/schema/DenseVectorField.java

```diff
@@ -193,12 +193,24 @@ public void checkSchemaField(final SchemaField field) throws SolrException {
     switch (vectorEncoding) {
       case FLOAT32:
         if (dimension > FloatVectorValues.MAX_DIMENSIONS) {
-          log.warn("The vector dimension "+ dimension +" specified for field "+ field.getName() +" exceeds the current Lucene default max dimension of "+ FloatVectorValues.MAX_DIMENSIONS +". It's un-tested territory, extra caution and benchmarks are recommended for production systems.");
+          if (log.isWarnEnabled()) {
```

Comment:
You don't need to add if statements to guard log.warn or log.error.

Reply:
Errorprone does tell you they are required sometimes. I just had to do it for one of my PRs.

Reply:
I am unaware Errorprone cares; can you provide a more specific reference please?

Our build has custom checks defined with Groovy in our Gradle build.  Maybe you meant that?  Erick Erickson was once burned by a debug log statement that had some arguments that were expensive to compute... and ever since he added these checks, we can no longer have the one-line convenience of a debug log message, not even in a test :-( (I'm not a fan TBH; cure was worse than what it was fixing IMO).    https://github.com/apache/solr/blob/7a9ce356bc7c90c8237437b1e16ed030e4d67900/gradle/validation/validate-log-calls.gradle#L110

Reply:
I didn't add the if check at the beginning but Gradle complained, do you believe I should ignore that and revert to the simple warn line?

Reply:
Admittedly I was just complaining about the "if" checks that I am opinionated about -- wasn't you.  The PR is fine :-)

Reply:
That said, if you have an interest/curiosity in why validate-log-calls.gradle is hassling you to do little things like this when we agree it should *not* (due to persnicketyLevel) -- I'd appreciate it.